packages/canvas/canvas-renderer/src/CanvasObjectRendererSystem.ts

```diff
@@ -0,0 +1,184 @@
+                renderTexture.valid = true;
+            }
+
+            _context.activeContext = renderTexture._canvasRenderTarget.context as CrossPlatformCanvasRenderingContext2D;
+            renderer.canvasContext.activeResolution = renderTexture._canvasRenderTarget.resolution;
+        }
+        else
+        {
+            _context.activeContext = _context.rootContext;
+            _context.activeResolution = rootResolution;
+        }
+
+        const context2D = _context.activeContext;
+
+        _context._projTransform = transform || null;
+
+        if (!renderTexture)
+        {
+            this.lastObjectRendered = displayObject;
+        }
+
+        if (!skipUpdateTransform)
+        {
+            // update the scene graph
+            const cacheParent = displayObject.enableTempParent();
+
+            displayObject.updateTransform();
+            displayObject.disableTempParent(cacheParent);
+        }
+
+        context2D.save();
+        context2D.setTransform(1, 0, 0, 1, 0, 0);
+        context2D.globalAlpha = 1;
+        _context._activeBlendMode = BLEND_MODES.NORMAL;
+        _context._outerBlend = false;
+        context2D.globalCompositeOperation = _context.blendModes[BLEND_MODES.NORMAL];
+
+        if (clear !== undefined ? clear : renderer.background.clearBeforeRender)
+        {
+            if (this.renderingToScreen)
+            {
+                context2D.clearRect(0, 0, renderer.width, renderer.height);
+
+                const background = renderer.background;
+
+                if (background.alpha > 0)
+                {
+                    context2D.globalAlpha = background.alpha;
+                    context2D.fillStyle = background.colorString;
+                    context2D.fillRect(0, 0, renderer.width, renderer.height);
+                    context2D.globalAlpha = 1;
+                }
+            }
+            else
+            {
+                renderTexture = (renderTexture as BaseRenderTexture);
+                renderTexture._canvasRenderTarget.clear();
+
+                const clearColor = renderTexture.clearColor;
+
+                if (clearColor[3] > 0)
+                {
+                    context2D.globalAlpha = clearColor[3] ?? 1;
+                    context2D.fillStyle = hex2string(rgb2hex(clearColor));
+                    context2D.fillRect(0, 0, renderTexture.realWidth, renderTexture.realHeight);
+                    context2D.globalAlpha = 1;
+                }
+            }
+        }
+
+        // TODO RENDER TARGET STUFF HERE..
+        const tempContext = _context.activeContext;
+
+        _context.activeContext = context2D;
+        displayObject.renderCanvas(renderer);
+        _context.activeContext = tempContext;
+
+        context2D.restore();
+
+        _context.activeResolution = rootResolution;
+        _context._projTransform = null;
+
+        renderer.emit('postrender');
+    }
+
+    public destroy(): void
+    {
+        // ka boom!
```

Comment:
Can you remove references here to lastObjectRenderer and renderer?

Reply:
done!